Error: `UA_STRING_NULL' can't be equated to common symbol 'UA_BYTESTRING_NULL'

Hi,

Using arm-buildroot-linux-musleabihf-gcc (gcc version 5.4.0 from Buildroot 2017.02.6) and your v0.2 release tag, compiling `src/ua_types.c` gives me the following error:
```
[ 87%] Building C object CMakeFiles/open62541-object.dir/open62541.c.o
{standard input}: Assembler messages:
{standard input}:32673: Error: `UA_STRING_NULL' can't be equated to common symbol 'UA_BYTESTRING_NULL'
CMakeFiles/open62541-object.dir/build.make:203: recipe for target 'CMakeFiles/open62541-object.dir/open62541.c.o' failed
make[4]: *** [CMakeFiles/open62541-object.dir/open62541.c.o] Error 1
```
I tracked this down to the use of the gcc flag `-fmerge-all-constants` in CMakeLists.txt.
I am able to fix the compilation error with this patch:
```diff
diff -u -r a/CMakeLists.txt b/CMakeLists.txt
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -136,7 +136,7 @@
     # Strip release builds
     if(CMAKE_BUILD_TYPE STREQUAL "MinSizeRel" OR CMAKE_BUILD_TYPE STREQUAL "Release")
         add_definitions(-ffunction-sections -fdata-sections -fno-stack-protector -fno-unwind-tables
-                        -fno-asynchronous-unwind-tables -fno-math-errno -fmerge-all-constants -fno-ident)
+                        -fno-asynchronous-unwind-tables -fno-math-errno -fno-ident)
         set(CMAKE_C_LINK_FLAGS "${CMAKE_C_LINK_FLAGS} -s")
         set(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} -s")
         if(APPLE)
```
Please comment on possibly integrating this patch into your source tree.
Thanks,
Vladimir

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -384,7 +384,7 @@ if(NOT UA_COMPILE_AS_CXX AND (CMAKE_COMPILER_IS_GNUCC OR "x${CMAKE_C_COMPILER_ID
     # Strip release builds
     if(CMAKE_BUILD_TYPE STREQUAL "MinSizeRel" OR CMAKE_BUILD_TYPE STREQUAL "Release")
         add_definitions(-ffunction-sections -fdata-sections -fno-stack-protector -fno-unwind-tables
-                        -fno-asynchronous-unwind-tables -fno-math-errno -fmerge-all-constants -fno-ident)
+                        -fno-asynchronous-unwind-tables -fno-math-errno -fno-ident)
         if(NOT OS9)
             set(CMAKE_C_LINK_FLAGS "${CMAKE_C_LINK_FLAGS} -s")
             set(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} -s")

